Generals/Code/GameEngine/Source/GameClient/MessageStream/SelectionXlat.cpp

```diff
@@ -1097,57 +1097,47 @@ GameMessageDisposition SelectionTranslator::translateGameMessage(const GameMessa
 					m_lastGroupSelTime = now;
 				}
 
-				// check for double-press to jump view
+				Drawable* draw = TheInGameUI->getFirstSelectedDrawable();
+				if (draw && draw->isKindOf(KINDOF_STRUCTURE))
+				{
+					//Kris: Jan 12, 2005
+					//Can't select other units if you have a structure selected. So deselect the structure to prevent
+					//group force attack exploit.
+					TheInGameUI->deselectAllDrawables();
+				}
 
-				if ( now - m_lastGroupSelTime < 20 && group == m_lastGroupSelGroup )
+				TheInGameUI->deselectAllDrawables(false); //No need to post message because we're just creating a new group!
+
+				Player* player = ThePlayerList->getLocalPlayer();
+				if (player)
 				{
-					DEBUG_LOG(("META: DOUBLETAP select team %d",group));
-					Player *player = ThePlayerList->getLocalPlayer();
-					if (player)
+					Squad* selectedSquad = player->getHotkeySquad(group);
+					if (selectedSquad != NULL)
 					{
-						Squad *selectedSquad = player->getHotkeySquad(group);
-						if (selectedSquad != NULL)
+						VecObjectPtr objlist = selectedSquad->getLiveObjects();
+						Int numObjs = objlist.size();
+						for (Int i = 0; i < numObjs; ++i)
 						{
-							VecObjectPtr objlist = selectedSquad->getLiveObjects();
-							Int numObjs = objlist.size();
-							if (numObjs > 0)
+							if (objlist[i]->getControllingPlayer() == player)
```

Comment:
This also looks like something that would be better suited for a Unify change? It looks like it fixes something.